### ISSUE TITLE
Migrate to Material-UI v5

### ISSUE THREAD

**Issue by Rugvip** [backstage/backstage#7094]
> [!IMPORTANT]
> Update: We're exploring https://github.com/backstage/backstage/issues/27726, and recommend holding any ongoing MUI 5 migrations until the path forward is clear.

Material-UI v5 is [on the way](https://material-ui.com/versions/#release-schedule) and will likely be a sizable [migration](https://next.material-ui.com/guides/migration-v4/). We want to make sure we make the migration as smooth as possible for everyone and avoid forcing migration at a particular point in time to whatever extent we can.

Let's use this issue as a point to collect and discuss ideas around how the migration can be done and when.

This is also likely to be very intertwined with #6910

**Comment by ellisio** [comment]
@Rugvip I know this kind of question is probably asked _a lot_, but when do you guys plan on going stable `@1.0.0`? Something like this screams post-1.0.0 since it is a very breaking type change. Especially for the metric ton of plugins people have developed.

**Comment by Rugvip** [comment]
@ellisio Realized I totally forgot to reply here, sorry about that 😅

Right now we're working towards stabilization and 1.0 of the frontend core, but are excluding the component library and theming, and with that MUI. Stabilization of those parts will be a slightly later effort, but I agree that MUI v5 migration is something that should be done before we reach 1.0 there.

**Comment by jwlawrence** [comment]
Beyond just planning around the 1.0 release, I think it's a good idea to have a plan for these migrations going forward, since there will be no doubt be a mui 6, 7, 8... It's an interesting issue since by having the backstage components depend on mui you take it off the market as a viable component library for those wishing to leverage something to build their own UI, unless they are ok with being pegged to whichever version backstage offers and migrating when backstage decides to.

Has any thought been given to migrating off mui as a dependency and creating custom components for backstage? I know that would present a huge amount of effort, but it might pay off in the long run in terms of avoiding churn for consumers.

In any case, I'll try to find time to help with this migration in whatever capacity I can.

**Comment by freben** [comment]
I'm not sure that that problem would be any better, or even significantly different, if we made our own component library. Leaning on MUI lets us leverage their investment and knowledge around accessibility, design, customizability and many other little things, as well as tap into the huge amount of third party component libraries out there that are built on top of it. We just do not have sufficient capacity or skill set, I think, to cover a design system of that scale. I'll humbly admit that I'm convinced that at least I, personally, would surely do a terrible job of it :)

**Comment by freben** [comment]
That being said, yes, you are right that this is a very important and wide contact surface that takes careful consideration. We are taking steps to ensure that the MUI dependencies are as walled off as possible, to try to keep that surface in check. But ultimately, most of the component library in the end will be using MUI semantics to some degree. Not necessarily directly - for example we have our Button and Link etc that shield things off somewhat even though that isn't their primary intent - but it's there.

**Comment by Rugvip** [comment]
@jwlawrence yep good point, and definitely a valid concern. The bump to v5 is not something we view as a one-off thing, but also as a learning opportunity for making future migrations as smooth as possible. I think one of the main things to explore here is the possibility of running multiple versions of MUI in parallel, with the intention being that that would be a temporary state that you're in during migration.

When it comes to moving to using our own component library, it's what freben said ^. I don't really think it's an area where Backstage or developer portals in general differ from other web applications. It's not until you get to a high level that there might be some components or patterns that are particularly common or well suited for Backstage, which is what `core-components` is about.

**Comment by yousifalraheem** [comment]
I'm really interested in this issue since our system is slowing down as it grows in complexity. MUI v5 offers a performance boost that is much needed. Some tests show more than 50% faster rendering.

**Comment by freben** [comment]
Small update on progress.

Some significant headway has been made toward enabling this upgrade eventually; MUI dependencies have been gradually moved out of core, and the [React 17](https://github.com/backstage/backstage/issues/6910) update is merged. So prerequisites are  falling into place.

**Comment by devth** [comment]
Hi @freben, what's next? Need any help from the community on this?

**Comment by freben** [comment]
@devth Hi! Thanks for offering - from our point of view, this is as important as it is hard. It is also not something that can be easily done in smaller pieces. It'll probably not be easy to delegate to the community at this point, but we are happy to hear of ideas or others' stories that could help us scope and plan for it a bit better. As it stands, we're considering that the maintainers would make it an explicit goal as part of some future planning, and dig down doing it as a concerted sprint effort.

**Comment by devth** [comment]
Makes sense. FWIW, I recently migrated a medium sized app from MUI 4 to 5 and it wasn't too painful. Their provided codemods worked quite well and handled most of the heavy lifting.

**Comment by webark** [comment]
Is there a "import patch" (not sure the right term) that maps the old imports to the new and ployfils some of the api changes? It the opposite that does the new to the old? Almost like a branching by abstraction mechanism?

I think the migration of the core packages and plugins seems doable, but it's the immediate breaking of existing applications and community plugins, and the inability to front or back load the changes it that make it challenging.

**Comment by webark** [comment]
Also, introducing a code mod step in the upgrade script could help facilitate this, as well as just general upgrade changes that would be useful as things change outside of this upgrade.

**Comment by haagenson** [comment]
Are there any updates to when Backstage will be upgraded to use MUI5?  We have some teams wanting to move their react components over to our installation of Backstage, but they were already written in MUI5.

**Comment by Rugvip** [comment]
To be honest the maintainer team likely won't have time to focus on this for the rest of the year. It's at the top of our migration list but got a lot of other things going on.

Any help would be appreciated, especially investigations into what needs to be done. For example, what happens if you try to use those MUI 5 components today? In what way do things explode? 😅

**Comment by webark** [comment]
I'd also like to see what the reverse is. (if backstage core was upgraded, how could mui4 components still work)

I'd be curious what the effort to get "backstage core/proper" upgraded would be using a code mod.

https://github.com/mui/material-ui/blob/master/packages/mui-codemod/README.md#-preset-safe

One of the more annoying things would be any inflight PRs breaking.

I'd say bolting on mui5 components onto a mui4 base without a clear timeline is a bit of a non starter. maybe for a shot interval, but not as any kind of long term prescription.

If a PR was made largely using the code mod and fixing any bugs, how fast would you be willing to move on that? Would needing existing instances to be able to upgrade without upgrading their components to mui5 be a requirement?

**Comment by Rugvip** [comment]
@webark We would very much want to run the migration similarly to other big migrations, like React Router stable, where all apps and plugins can migrate at their own pace. If needed we still might set some form of deadline where we would drop support for MUI v4, but that'd likely be quite far in the future.

**Comment by waldirmontoya25** [comment]
Wouldn't it be helpful to go back to the basics as part of the migration and use MUI across the UI so that we can take advantage of the docs, patterns, storybook, design guide, etc.? Wouldn't it make theming and other design initiatives easier?

**Comment by webark** [comment]
@waldirmontoya25 what do you mean? MUI is what is used for all the UI.

**Comment by waldirmontoya25** [comment]
@webark I meant just use MUI components. What we have seen is that not all components adhere to MUI's guidelines. Below the summary of the pain points identified:

**Documentation:** Team designers, as well as developers, struggle to understand what UI components are available to them and what the latest versions of those components look like because there are conflicting sources of information (MUI docs, Backstage Storybook, Backstage codebase, Backstage Figma file)

**Discoverability:** Waste of time reinventing the wheel because they didn’t realize the UI component they needed already existed elsewhere (e.g., in another plugin)

**Quality:** Some of the custom Backstage components that have been documented in Backstage’s Storybook have less functionality than similar components that MUI has already defined (e.g. alerts)

**Accountability:** there is currently little incentive for contributors to invest in accessibility and visual design quality before merging changes (when we first started contributing to Backstage, our accessibility audit returned 112 issues)

**Theming:** Hardcoded values and other deviations from MUI’s theming engine prevent entire Backstage instances from being easily themed out-of-the-box (e.g., all the buttons might turn orange except for one that stays blue)

**Comment by luchillo17** [comment]
Hi there, in the #14671 PR we're trying to sync the core components to standardize the usage of MUI (plugins tests we're updated accordingly, but each plugin should adhere to the [Component Design Guidelines](https://backstage.io/docs/dls/component-design-guidelines) in a different PR), hopefully this helps with the issue mentioned before about components not adhering to the guidelines, as well as make migration easier.

At VMWare we're interested on the V5 upgrade work, based on this issue's discussion the main points would be:

- **Partial upgrade**: Can't find any docs around something like this, in fact I would consider it a bad thing, what if we forget to update a part of the codebase because how big it is? more so with the codemods officially setup by MUI to handle the migration, it should be simpler to just run the codemods on the whole project, fix the issues and then instruct the plugin authors/maintainers to refactor the result to make sure the theme still looks ok, then instruct any Backstage instance admin to run the codemods and fix any custom implementations they have.

- **Codemods**: We should rely on MUI codemods as much as possible, then if we see there's a repeatable pattern of fixes we(maintainers, plugin authors or users) have to apply after MUI codemods, then we can think of creating custom codemods (if any necessary) to bridge the upgrade process.

- **V4 Theme**: @webark mentioned V4 & V5 themes being incompatible, however there seems to exist an [adaptV4Theme helper](https://mui.com/material-ui/migration/v5-style-changes/#%E2%9C%85-add-adaptv4theme-helper) which should bridge that at the start, off course we should eventually remove the helper to use a V5 compatible theme.

- **Tests**: As far as I've seen while working with `@backstage/core-components` the tests do not rely on anything specific to MUI, we query components by text content (as recommended by React), and that will pay out in this update, I would expect minimal changes to tests in the update, any issues in tests would happen around build issues after running the codemods.

- **Storybook**: The Storybook has been outdated for some time, some components do not have stories, and those that do are overly simplistic, I want to note having a well supported storybook would help on checking the before and after update to make sure the update didn't introduce UI changes that the tests can't catch, since we could compare the [Backstage's Figma](https://backstage.io/docs/dls/figma) page with the actual implementation in the [Storybook](https://backstage.io/docs/dls/contributing-to-storybook).

- **Core & Plugins but no Instances**: Stand to reason we should not involve ourselves with specific user instances other than pointing to MUI upgrade docs & providing custom codemods (if needed), the question here would be if a first PR should only care about the core packages or should we also target the plugins, feel like other than the codemod run  we should not touch plugins as long as the tests are running properly, we (or any other maintainer) would lack context in each plugin to apply any refactors or fixes.

Am I missing anything?

**Comment by webark** [comment]
I was going to go back and try to fiddle with it, but the theme only partially came over. Also the way that you reference nested classes changed as well, and so did the override mechanism. I didn't spend any time fiddling with that, but only getting all the deps correct and trying to get the tests passing.

**Comment by luchillo17** [comment]
You mean CSS classes for nested components right? so the codemods weren't able to change the class nesting and overrides properly? do you have a branch to look at? I'm going to try myself tomorrow and see the issues pop up.

**Comment by webark** [comment]
Yea. This https://mui.com/material-ui/migration/v5-style-changes/#migrate-theme-styleoverrides-to-emotion

**Comment by webark** [comment]
And i didn't push a branch. I just have it on my local still

**Comment by devth** [comment]
+1 for not wrapping MUI.

The MUI project has incredibly high standards, momentum, and a massive community to support high quality, consistency, up-to-date docs, storybooks, tests, and thoughtful APIs that expose the right controls to the end user.

**Comment by luchillo17** [comment]
A year ago people were discussing doing this before 1.0, that's already shipped off so, are we going to mark this for a 2.0 release?

**Comment by webark** [comment]
In regards to not wrapping MUI components, what components are you talking about specifically? Things like the "info card" or the layout ones like "page" and "header"? Or like Links and buttons..

**Comment by luchillo17** [comment]
As far as I see (can't be sure all the `@backstage/core-components` follow this) Backstage doesn't wrap the MUI components, it presents reusable abstractions (DRY principle) that Backstage expect plugin developers & app admins to use more frequently (be it only style abstractions or feature ones).

However Backstage doesn't forbid the use of MUI components, it recommends using the core components whenever possible, and then we can use the MUI components if none core ones fit our needs [plugin-development#developing-guidelines](https://backstage.io/docs/plugins/plugin-development#developing-guidelines).

Case in point `InfoCard` component does use the MUI card under the hood, but adds styles as well as some extra features like `ErrorBoundary` and `BottomLink`, so I wouldn't say `InfoCard` is a wrapper of MUI's `Card`.

**Comment by webark** [comment]
There are a chunk of the core components that just have a storybook in it. Maybe that's what you are mentioning?

**Comment by luchillo17** [comment]
There are some that don't, there are some that have a single story without any kind of support for [Storybook Controls](https://storybook.js.org/docs/react/essentials/controls) nor [Storybook Actions](https://storybook.js.org/docs/react/essentials/actions), and then there are some with either partial or proper support.

**Comment by luchillo17** [comment]
Tried the update, steps are simple but the sheer amount of changes is hard to deal with, more than 1000 file changes (mostly thanks to the MUI codemods & `yarn workspaces foreach` to change deps in all packages & plugins), the issues are the breaking changes that the codemods can't handle, checking & fixing the build errors just takes a lot of time...

**Comment by testn** [comment]
Do we have a way going forward with this?

**Comment by webark** [comment]
not completely? I kind of want to find a window sometime, where we can kind of pause the work on the rest of the front end plugins and packages, and put focus on getting this out. Its just becoming more and more of a wart. It's just going to be very disruptive from a files changed perspective, and i'm not sure on the feasibility of having both be used simultaneously. 

There are some places that we could do some prep work to get ready for it. I think @luchillo17 did some work recently in that area.

**Comment by Rugvip** [comment]
I spent a bit of time towards this to see if we could get it kickstarted. Proposing a way forward for this in #15484. I think we can get simultaneous v4 + v5 support going, and to be honest we really need to be able to make this an incremental migration. There's way too much potential work to migrate to v5 that we can't force it as part of a single release. I do think we want to make it somewhat fast though, within 6 months or maybe less. The TypeScript performance seems to take a pretty big hit when running v4 and v5 at the same time. It's either that or we figure out a way for projects to completely drop v4 when they can.

**Comment by webark** [comment]
Thanks so much for working on this! I know this whole process is a bit of a drag all around :/ 

Is there a way we could just have an optional theme provider for those that want to have both v4 and v5 running simultaneously? Most consumers could probably handle the migration within a couple of weeks as the updates after running the code modes are pretty light aside from the theme work. And having the unified theme be an opt in would give more flexibility for upgrading? 

And are you saying upgrade the backstage repo within 6 months? or have the option of running bother only be supported for 6 months.

Thank you again. I know these kinds of situations there's no great solution.. :/

**Comment by Rugvip** [comment]
@webark i think the most sensible option is gonna be to make v4 support optional, based on whether v4 is present or not. I'll likely take a stab at that once I spend some more time on this. Basically if v4 isn't installed we won't bother creating a theme for it.

Regardless of how fast people upgrade their own projects there's going to be plenty of usage of other OSS plugins that need to be migrated too. 

The 6 months would be the window that we support v4 at all, although tbh if that support is lightweight and out of the way we can probably keep it for longer.

**Comment by webark** [comment]
Yea. community plugins are the long tail effort in my opinion. As long as backstage proper doesn't get stuck in the limbo of both, then I think this plan is 🤩

**Comment by luchillo17** [comment]
We should make a distinction between supporting MUI v4 & supporting MUI v4 theme, there's a theme adapter so a V4 theme can be used with MUI v5, @Rugvip when you say v4 + v5 support you mean the themes or the actual MUI codebase & components?

**Comment by oliviertassinari** [comment]
I had a quick look at this migration GitHub issue and related PR. What I noticed:

1. React 17 was a friction point. It's now [solved](https://github.com/backstage/backstage/issues/7094#issuecomment-1017856838). Maybe MUI shouldn't have made it a requirement https://github.com/mui/material-ui/pull/35232#discussion_r1030906426.
2. Being able to run v4 and v5 at the same time is critical for Backstage. Both because it's impossible to migrate everything to a working state in **one** PR and because of the community plugins that need a smooth upgrade path, a la WordPress. On this, it seems that MUI could improve:

  - 1. The migration docs, and document how to run v4 and v5 in the same codebase. using https://mui.com/x/react-data-grid/migration-v4/#using-mui-core-v4-with-v5 as inspiration. It should be before any of the imports are updated https://mui.com/material-ui/migration/migration-v4/#replace-all-imports. The current migration v5 docs only works with small projects.
  - 1. Expose backward compatibility adapters. I saw the opposite adapter to [`adaptV4Theme`](https://mui.com/material-ui/migration/v5-style-changes/#%E2%9C%85-add-adaptv4theme-helper) was created. It's something MUI could expose (using a v5 theme with v4 components).
  
Anything else?

**Comment by Rugvip** [comment]
@luchillo17 @testn v4 + v5 support means both theme as well as codebase and components, i.e. you should be able to use components from for example `@material-ui/core` and `@mui/material` in the same app. I've spent some time digging into different ways of loading in the different MUI versions as weak dependencies in the theme package. It adds a bit of complexity but it might be something we end up going with. There are a couple of different points in the migration where it would be beneficial to support weak deps:

- Phase 1: Initial MUI v5 support
  - In this first step we add the ability to use the v5 theme, and to start using v5 components. At this point it would be beneficial to have MUI v5 be a weak dependency in order to only add the overhead for users that want to get out ahead of the migration.
- Phase 2: Simultaneous v4 and v5 support
  - The next step is to start migrating all Backstage packages (both `@backstage/*` and external) to MUI v5. At this point we will require support for both MUI v4 and v5. We won't aim to migrate all packages within a single release cycle. There overhead of running both versions will be present in all apps that upgrade to these versions.
- Phase 3: Optional v4 support
  - Once all packages used within a given app have been migrated to MUI v5, it should be possible to drop support for MUI v4 in that app. At this point we want it to be possible to have MUI v4 be a weak dependency in order to not add the overhead of running MUI v4. Once a given app reaches this state it has completed the MUI v5 migration.

I'd say that the main thing to consider is whether we want to try to make MUI v5 optional during Phase 1, as this would avoid the overhead of v4 + v5 for one release cycle (1 month). I've run some simple benchmarks to get an idea of what the overhead is like for adding MUI v5 to an existing app with MUI v4:

- `example-app` in this repo:
  - `yarn build`: `29.541s +1.79s ~6.44%`
  - `yarn tsc:full`: `58.376s +3.38s ~6.14%`
- Newly created app from `@backstage/create-app`:
  - `yarn build`: `18.085s +0.57s ~3.25%`
  - `yarn tsc:full`: `11.743s +1.66s ~16.44%`

I feel that these numbers are fairly reasonable, as in I don't think it's worth adding a lot of complexity in order to avoid this overhead for one release cycle. If we can achieve it with relative easy without complicating phase 2 it's probably worth it though.

**Comment by Rugvip** [comment]
@oliviertassinari Thank you for having a look! 🙏

Regarding React 17 we support both v16 and v17 right now, although we've been shipping v17 as the default for a long time now, and only test towards v17. Given that we also want to move on to v18 we'll likely drop support for v16 soon. It would be nice not to be forced to do that as part of the MUI migration though, as it stacks two migrations on top of each other. We can of course drop v16 support first though, perhaps even as part of phase 1, mentioned above.

👍 for updating migration docs to mention simultaneous use of v4 and v5, I did look around quite a bit for that and couldn't find those docs.

Same thing for the v5 -> v4 adapter, would be sweet to have an official one.

There is one more thing I ran into, which is https://github.com/mui/material-ui/issues/35287. The workaround is baked into the v5 types, but it's always a bit ugly to ship those kinda global types: https://github.com/backstage/backstage/blob/bd63a210c19359c4307b91d08e0b975b151e5936/packages/theme/src/v5/types.ts#L39-L48
I'd wonder if there's something that could be done with conditional types in MUI v5 to avoid the issue?

**Comment by Pike** [comment]
Phase 2 would also be the timeframe for public plugins to migrate, right?

If they wanted to support both v4 and v5, they'd need some sort of dot releases, and release a v4 and a v5 independently?

Might be good to suggest some best practices for this?

**Comment by Rugvip** [comment]
@Pike yep, I don't expect plugins themselves to have the need for any form of simultaneous support though. Once we guarantee that v5 theming is available in the app, any plugin would be free to migrate to v5. Whether a plugin is partially of fully migrated hopefully shouldn't be visible from the outside. It simply uses either v4 or v5 components internally.

**Comment by siriwatknp** [comment]
> * Phase 1: Initial MUI v5 support
>   
>   * In this first step we add the ability to use the v5 theme, and to start using v5 components. At this point it would be beneficial to have MUI v5 be a weak dependency in order to only add the overhead for users that want to get out ahead of the migration.
> * Phase 2: Simultaneous v4 and v5 support
>   
>   * The next step is to start migrating all Backstage packages (both `@backstage/*` and external) to MUI v5. At this point we will require support for both MUI v4 and v5. We won't aim to migrate all packages within a single release cycle. There overhead of running both versions will be present in all apps that upgrade to these versions.

@Rugvip Would it be possible if phases 1 and 2 are combined and provide an escape hatch for people who don't need v4?

**Comment by waldirmontoya25** [comment]
@Rugvip So, I guess we have to wait until https://github.com/backstage/backstage/pull/15484 gets merged so we can start contributing to plugin migration, correct?

**Comment by Rugvip** [comment]
@waldirmontoya25 yep!

**Comment by webark** [comment]
https://github.com/material-table-core/core/releases/tag/v6.0.0 :/

**Comment by benjdlambert** [comment]
@webark yep we saw this coming and it's another motivation for us to push the muiv5 migration. Hoping to get this prioritized H1 this year.

**Comment by oliviertassinari** [comment]
@Rugvip Thanks for the feedback on what Material UI could do to improve the upgrade experience. I had a look at t he type issue, I proposed a solution in: https://github.com/mui/material-ui/issues/35287#issuecomment-1408870399.

**Comment by Rugvip** [comment]
The customization issue highlighted in #16631 should be taken into consideration for this. In particular if we decide to change or add recommendations for how components are styled, we should make sure those recommendations are compatible with the theming system.

**Comment by awanlin** [comment]
Hi @Rugvip, I could swear I saw somewhere mentioned that this was going to be worked on for the June release - `1.15.0` - can you confirm or is that just my imagination?

We have some desire to use our internal components but they are all on MUI v5 and because of that have some resources that would be able to help

**Comment by Rugvip** [comment]
@awanlin yep @tudi2d has spent some time to get #15484 in a shippable state, as well as a few followup migrations. If all goes well we'll ship it into `next` straight after the main line release next week

**Comment by awanlin** [comment]
Awesome thanks for the update @Rugvip and @tudi2d for pushing it forward some more.

**Comment by ypetrych** [comment]
Hi guys! Could you provide some specific date when you plan to release the migration? Thanks a lot!

**Comment by awanlin** [comment]
Hi @ypetrych, given all the things still in-progress and summer vacation it would not be wise for us to share a date or tentative release for this. We should be better able to answer this towards the end of summer. One part that is slowing things down is getting the React 18 upgrade completed. There's an RFC around that you might want to read: https://github.com/backstage/backstage/issues/18625

That said, if is important to you then if you are on version `1.15.0` or newer you can do this already. You need to upgrade to at least React 17 but I'd recommend going to React 18. From there you can update to using Material UI v5 following their migration guide: https://mui.com/material-ui/migration/migration-v4/. Keep in mind that there may be issues that we have no encountered yet. The organization I work for has done this and so far it's been solid for us but we are a mid-sized so there's a chance we did not hit some unknown pain points. I would say the part that required the most work was fixing and updating tests.

**Comment by webark** [comment]
Is there existing documentation around the need to use the `v5-` class prefix for overriding styles? Or are we using the local overrides wrong and shouldn't be needing this.

**Comment by webark** [comment]
@tudi2d Is it the same for overwriting sub items at a component level?

When using the "sx" property, we had to add the "v5-" for them to get picked up. Should we not have to even when it's individual components? (not at the theme, at the ex property)

**Comment by LegendSebastianoL** [comment]
now that users can migrate to `mui` V5 what about the `sx` support and finally remove the `makestyle`

**Comment by drodil** [comment]
So, this issue is quite old and there's already MUIv6 available. Is the migration going to be from 4->5 or straight from 4->6? And are there any timeline for this, perhaps in relation to the new frontend system?

**Comment by webark** [comment]
From what have heard they are focusing on the Migrating completely to the new backend and front end first, Then this will be a cleaner migration.

We use a lot of MUI 5 components, and are at mui 5.16 currently without issue. The main annoyance for us is the material table core, but we are thinking about redoing our main tables with a more modern library that works either graphql better.

**Comment by Rugvip** [comment]
We're actually looking into this atm 😁, goal is to make sure that MUI v5 and v6 adoption is properly supported, bugs are ironed out, and with a path to completely be able to remove MUI v4 at some point. For now the best place to track it is in the MUI v5 milestone still, linked in this issue.

**Comment by deejay1** [comment]
One issue we found is that MUI 5 supports react >= 17, while there's a check that peer deps in main include react 16. Should this be changed to allow migrating plugins?

**Comment by Rugvip** [comment]
This is no longer a goal, https://github.com/backstage/backstage/issues/27726, see https://ui.backstage.io
